### ISSUE TITLE
Shell Error 127

### ISSUE THREAD

**Issue by siicockerill** [ChrisBraybrooke/LaravelChromePdf#3]
I'm getting a shell error 127 in my live environment when using your package.

In the **troubleshooting section** of the https://github.com/spiritix/php-chrome-html2pdf:
> Manually set the path to your Node.js executable using the $converter->setNodePath() method

How can I do that via your package? Is there a method?

**Comment by ChristianBraybrooke** [comment]
Hi @siicockerill,

There isn't currently a method to set the node path - but I can look into adding this. What is shell error 127? Is it definitely related to the node path?

Thanks, Christian

**Comment by siicockerill** [comment]
Hi @ChrisBraybrooke 👋  

Here's a related issue — https://github.com/spiritix/php-chrome-html2pdf/issues/6#issue-362261804

What  I'm experiencing isn't definitely an issue related to setting the node path itself. It's possibly a permissions issue. But, I've been debugging this for hours and not being able to set the path via your package, stops me from eliminating it as an issue.

The package works brilliantly in my local environment 👏

**Comment by ChristianBraybrooke** [comment]
Yeah that makes sense, I can probably get this added at some point over the weekend.

Out of interest - what environment are you trying to run it in?

**Comment by siicockerill** [comment]
Ubuntu 14.04.5 LTS

That would be a great help! 👍

**Comment by siicockerill** [comment]
Hi Chris,

Update for you — I was able to get puppeteer to install following the instructions over on the Spatie Browsershot package link: https://github.com/spatie/browsershot

```
curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -
sudo apt-get install -y nodejs gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget
sudo npm install --global --unsafe-perm puppeteer
sudo chmod -R o+rx /usr/lib/node_modules/puppeteer/.local-chromium
```
